TesseractOCR/G8Tesseract.mm

```diff
@@ -310,18 +340,124 @@ - (void)setVariablesFromDictionary:(NSDictionary *)dictionary
 
 - (void)loadVariables
 {
-    [self.variables enumerateKeysAndObjectsUsingBlock:^(NSString *key, NSString *value, BOOL *stop) {
-        _tesseract->SetVariable(key.UTF8String, value.UTF8String);
-    }];
+    if (self.isEngineConfigured) {
+        [self.variables enumerateKeysAndObjectsUsingBlock:^(NSString *key, NSString *value, BOOL *stop) {
+            _tesseract->SetVariable(key.UTF8String, value.UTF8String);
+        }];
+    }
+}
+
+#pragma mark - Internal getters and setters
+
+- (tesseract::TessBaseAPI *)tesseract {
+    
+    if (!_tesseract) {
+        _tesseract = new tesseract::TessBaseAPI();
+    }
+    return _tesseract;
+}
+
+- (void)setEngineImage:(UIImage *)image {
+    
+    if (image.size.width <= 0 || image.size.height <= 0) {
+        NSLog(@"ERROR: Image has not size!");
+        return;
+    }
+    
+    image = [image fixOrientation];
+    
+    self.imageSize = image.size; //self.imageSize used in the characterBoxes method
+    
+    if (self.isEngineConfigured) {
+        Pix *pix = nullptr;
+        
+        if ([self.delegate respondsToSelector:@selector(preprocessedImageForTesseract:sourceImage:)]) {
+            UIImage *thresholdedImage = [self.delegate preprocessedImageForTesseract:self sourceImage:image];
+            if (thresholdedImage != nil) {
+                self.imageSize = thresholdedImage.size;
+                
+                Pix *pixs = [self pixForImage:thresholdedImage];
+                pix = pixConvertTo1(pixs, UINT8_MAX / 2);
+                pixDestroy(&pixs);
+                
+                if (pix == nullptr) {
+                    NSLog(@"WARNING: Can't create Pix for custom thresholded image!");
+                }
+            }
+        }
+        
+        if (pix == nullptr) {
+            pix = [self pixForImage:image];
+        }
+        
+        @try {
+            _tesseract->SetImage(pix);
+        }
+        //LCOV_EXCL_START
+        @catch (NSException *exception) {
+            NSLog(@"ERROR: Can't set image: %@", exception);
+        }
+        //LCOV_EXCL_STOP
+        pixDestroy(&pix);
+    }
+    
+    _image = image;
```

Comment:
I don't think we should save the image if the engine isn't configured or if we're not successful in calling `_tesseract->SetImage(pix)`. Can you rework this function so that's not the case?